docs/scratch.rst

```diff
@@ -34,3 +34,25 @@ For example:
         Assert(myvar.load() == Int(5))
     ])
     anotherVar = ScratchVar(TealType.bytes, 4) # assign this scratch slot to slot #4
+
+DynamicScratchVar:  Referencing a ScratchVar
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:any:`DynamicScratchVar` functions as a pointer to a :any:`ScratchVar` instance.
+
+Reference a :any:`ScratchVar` instance by invoking :any:`DynamicScratchVar.set_index`.  :any:`DynamicScratchVar.set_index` *must* be invoked before using :any:`DynamicScratchVar.load` and :any:`DynamicScratchVar.store`.
```

Comment:
If we're going to warn the user about calling `load/store` before `set_index`, we should also warn users in the regular `ScratchVar` case not to call `load` before `store`. For regular `ScratchVar`'s there is an exception at compile-time, but I don't believe we catch the `DynamicScratchVar` issue you just identified (but I haven't checked this)

Reply:
Here's where the ["load occurs before store"](https://github.com/algorand/pyteal/blob/b261fa55a35841c3e8b1878cac868c19ae905ed6/pyteal/ir/tealblock.py#L108) error is called during compilation

Reply:
@tzaffi Thanks for the pointer.  Sorry that I missed it in my brief look.  Via https://github.com/algorand/pyteal/pull/226/commits/83027e2c25efb748cef127443a62905d43d6e9c1, I updated docs to reflect the prerequisite.